apps/nextjs-website/src/components/organisms/WebinarQuestionsForm/WebinarQuestionsForm.tsx

```diff
@@ -1,39 +1,62 @@
+'use client';
+import { snackbarAutoHideDurationMs } from '@/config';
+import { addWebinarQuestion } from '@/helpers/webinarQuestions.helpers';
+import { DevPortalUser } from '@/lib/types/auth';
 import { Done } from '@mui/icons-material';
 import { LoadingButton } from '@mui/lab';
-import { Card, TextField, Typography } from '@mui/material';
+import { Alert, Card, Snackbar, TextField, Typography } from '@mui/material';
 import { useTranslations } from 'next-intl';
-import { useState } from 'react';
-
-const sendQuestion = (question: string) => {
-  return new Promise((resolve) => {
-    setTimeout(() => {
-      resolve('ok');
-    }, 3000);
-  });
-};
+import { useCallback, useState } from 'react';
 
 type FormState = 'submitting' | 'submitted' | undefined;
 type WebinarQuestionsFormProps = {
   disabled?: boolean;
+  webinarSlug: string;
+  user: DevPortalUser;
 };
 
 export const WebinarQuestionsForm = ({
   disabled = false,
+  webinarSlug,
+  user,
 }: WebinarQuestionsFormProps) => {
   const t = useTranslations('webinar');
+  const [error, setError] = useState<string | null>(null);
   const [question, setQuestion] = useState('');
   const [formState, setFormState] = useState<FormState>();
 
+  const sendQuestion = useCallback(async () => {
+    return await addWebinarQuestion({
+      email: user.attributes.email,
+      givenName: user.attributes.given_name,
+      familyName: user.attributes.family_name,
+      question: question,
+      webinarSlug: webinarSlug,
+      data: new Date().toISOString(),
+    });
+  }, [webinarSlug, user, question]);
+
+  const handleError = () => {
+    setFormState(undefined);
+    setError(t('questionsForm.error'));
+  };
+
   const handleSubmit = () => {
     if (!question) return;
     setFormState('submitting');
-    sendQuestion(question).then(() => {
-      setFormState('submitted');
-      setQuestion('');
-      setTimeout(() => {
-        setFormState(undefined);
-      }, 3000);
-    });
+    sendQuestion()
+      .then((res: { status: string }) => {
```

Comment:
Let the compiler infer the type. It should be determined by the `addWebinarQuestion` function (add the return type there). For now you can skip the validation of the response (e.g. using `io-ts`), because all this code is going to be replaced soon.

Reply:
I removed explict type and move to `addWebinarQuestion` all the error mangement relative to response. For now i manage response here using standard js `Promise` then will be refactored using `io-ts` when `addWebinarQuestion` will be proper implemented